src/annotator/config/index.js

```diff
@@ -1,12 +1,57 @@
 import settingsFrom from './settings';
 import { toBoolean } from '../../shared/type-coercions';
 
+/**
+ * Total list of configuration keys.
+ */
+const configurationKeys = [
+  'annotations',
+  'assetRoot',
+  'branding',
+  'clientUrl',
+  'enableExperimentalNewNoteButton',
+  'experimental',
+  'group',
+  'focus',
+  'theme',
+  'usernameUrl',
+  'onLayoutChange',
+  'openSidebar',
+  'query',
+  'requestConfigFromFrame',
+  'services',
+  'showHighlights',
+  'notebookAppUrl',
+  'sidebarAppUrl',
+  'subFrameIdentifier',
+  'externalContainerSelector',
+];
+
+/**
+ * List of allowed configuration keys per application context. Keys omitted
+ * in a given context will be removed from the relative configs when calling
+ * getConfig.
+ */
+const configurationContexts = {
+  // For testing
+  all: [...configurationKeys],
+
+  annotator: [...configurationKeys],
+
+  sidebar: [...configurationKeys],
```

Comment:
Is there a need to clone the list? Since we're not mutating it you could just do `sidebar: configurationKeys`.